### PR TITLE
Separate statements from GenTree, part 1

### PR DESCRIPTION
**Pull request by sandreenko**
bcbf3704a8: Fix some leftovers where we used GenTree instead of GenTreeStmt.

39adc78439: Separate GT_IL_OFFSET and GT_STMT.

1c50f5b2b9: Add unreached to gtHashValue for GT_STMT.

it has an assert that tree is not GT_STMT: https://github.com/dotnet/coreclr/blob/65a50887b9a460d7833f0ab476fab9027f1e8166/src/jit/gentree.cpp#L1837 the assert was added before 2010.

3fff1716a9: Do not use gtCloneExpr for stms.

Contributes to #25899

### CODE REVIEW

**Comment by sandreenko** [comment]
PTAL @CarolEidt @dotnet/jit-contrib

File: src/jit/optimizer.cpp [hunk 1]
```diff
@@ -3703,9 +3694,9 @@ void Compiler::optUnrollLoops()
                     ++loopRetCount;
                 }
 
-                /* Visit all the statements in the block */
+                // Visit all the statements in the loop.
 
-                for (GenTreeStmt* stmt = block->firstStmt(); stmt; stmt = stmt->gtNextStmt)
+                for (GenTreeStmt* stmt = loopStmt; stmt != nullptr; stmt = stmt->gtNextStmt)
```
Comment (BruceForstall) [inline src/jit/optimizer.cpp]: Can you explain why this is correct, and why we shouldn't continue to use `block->firstStmt()` here?

**Comment by sandreenko** [comment]
> Have you verified there are no diffs?

I have checked x64 checked framework assemblies and there were no diffs, but looks like I need to check it again and probably with pmi.

File: src/jit/optimizer.cpp (hunk 1 above)
Comment (sandreenko) [reply src/jit/optimizer.cpp]: I can't. Good catch, thank you.

 Looks like `loopStmt` (or `loop` before this change) is a dead variable. Our loop includes all blocks from head to bottom and that is what we are trying to sum here.

I will fix it shorty.

**Pull request merged**